src/ProxyClient/LiteSpeed.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+
+/**
+ * LiteSpeed Web Server (LSWS) invalidator.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class LiteSpeed extends HttpProxyClient implements PurgeCapable, TagCapable, ClearCapable
+{
+    private $headerLines = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->headerLines[] = 'X-LiteSpeed-Purge: *';
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge($url, array $headers = [])
+    {
+        $this->headerLines[] = 'X-LiteSpeed-Purge: '.$url;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions()
+    {
+        $resolver = parent::configureOptions();
+
+        $resolver->setRequired(['target_dir']);
+        $resolver->setAllowedTypes('target_dir', 'string');
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function invalidateTags(array $tags)
+    {
+        $this->headerLines[] = 'X-LiteSpeed-Purge: '.implode(', ', preg_filter('/^/', 'tag=', $tags));
```

Comment:
if a lot goes wrong, could we end up with php server injection with a carefully crafted tag? i think we need some escaping here

Reply:
The invalidation request comes from the `CacheInvalidator`. This is trusted to me?

Reply:
thanks!